src/status_im2/contexts/wallet/events.cljs

```diff
@@ -311,6 +312,43 @@
                :wallet/valid-ens-or-address?
                false)}))
 
+(rf/reg-event-fx :wallet/find-ens
+ (fn [{:keys [db]} [input contacts chain-id cb]]
+   (let [result (if (empty? input)
+                  []
+                  (filter #(string/starts-with? (or (:ens-name %) "") input) contacts))]
+     (if (and input (empty? result))
+       (rf/dispatch [:wallet/search-ens input chain-id cb ".stateofus.eth"])
+       {:db (assoc db
+                   :wallet/local-suggestions
+                   (map #(assoc % :type item-types/saved-address) result)
+                   :wallet/valid-ens-or-address? (not-empty result))}))))
+
+(rf/reg-event-fx :wallet/search-ens
+ (fn [_ [input chain-id cb domain]]
+   (let [ens (if (string/includes? input ".") input (str input domain))]
+     {:fx [[:json-rpc/call
+            [{:method     "ens_addressOf"
+              :params     [chain-id ens]
+              :on-success #(rf/dispatch [:wallet/set-ens-address % ens])
+              :on-error   (fn []
+                            (if (= domain ".stateofus.eth")
+                              (rf/dispatch [:wallet/search-ens input chain-id cb ".eth"])
+                              (do
+                                (rf/dispatch [:wallet/set-ens-address nil ens])
+                                (cb))))}]]]})))
+
+(rf/reg-event-fx :wallet/set-ens-address
+ (fn [{:keys [db]} [result ens]]
+   {:db (assoc db
+               :wallet/local-suggestions     (if result
```

Comment:
We could do this check on the `on-success` of the `ens_addressOf` RPC call and dispatch a different event to set the `:wallet/valid-ens-or-address?` false. WDYT?

Reply:
I think we need to update the `:wallet/local-suggestions` anyways. If the result is empty, then we update with an empty array (to overwrite the previous suggestions)